src/renderer/RawRenderer.ts

```diff
@@ -2,10 +2,202 @@
  * Copyright (c) 2015 NAVER Corp.
  * egjs projects are licensed under the MIT license
  */
-import Renderer from "./Renderer";
-
+import Flicking from "~/Flicking";
+import Renderer from "~/renderer/Renderer";
+import ElementPanel from "~/core/ElementPanel";
+import { getFlickingAttached, getMinusCompensatedIndex, includes, parseElement, toArray } from "~/utils";
+import { ElementLike } from "~/type/external";
 class RawRenderer extends Renderer {
+  protected _panels: ElementPanel[];
+
+  public init(flicking: Flicking): this {
+    super.init(flicking);
+    this._collectPanels();
+    return this;
+  }
+
   public render() { return this; }
+
+  public insert(index: number, element: ElementLike | ElementLike[]): ElementPanel[] {
+    const panels = this._panels;
+    const flicking = getFlickingAttached(this._flicking, "Renderer");
+
+    const { camera, control } = flicking;
+    const align = this._getPanelAlign();
+
+    const elements = parseElement(element);
+    const insertingIdx = getMinusCompensatedIndex(index, panels.length);
+
+    const panelsPushed = panels.slice(insertingIdx);
+    const newPanels = elements.map((el, elIdx) => new ElementPanel({ el, index: insertingIdx + elIdx, align, flicking }));
+
+    panels.splice(insertingIdx, 0, ...newPanels);
+
+    // Update panel indexes & positions
+    panelsPushed.forEach(panel => {
+      panel.increaseIndex(newPanels.length);
+      panel.resize();
+    });
+
+    // Reset the order of the elements
+    this.resetPanelElementOrder();
+
+    // Insert the actual elements as camera element's children
+    const cameraElement = camera.element;
+    const nextSiblingPanel = panelsPushed[0]?.element || null;
+    const fragment = document.createDocumentFragment();
+
+    newPanels.forEach(panel => fragment.appendChild(panel.element));
+    cameraElement.insertBefore(fragment, nextSiblingPanel);
+
+    // Resize the newly added panels
+    newPanels.forEach(panel => panel.resize());
+
+    // Update camera & control
+    camera.updateRange();
+    camera.updateAnchors();
+    camera.resetNeedPanelHistory();
+    control.updateInput();
```

Comment:
This group seems to work as a combination. How about making it with a method?

Reply:
I'm not rushing to do it because there were a few changes in that code which makes it different from other parts.
I think merging them only makes it hard to change for future updates.
As it's a development phase for the v4, I think we can merge it later after the codes are stable.

Reply:
If so, I would like to leave a record somewhere.

Reply:
Sure, I'll leave a comment.